framework/aster-frame/src/task/scheduler.rs

```diff
@@ -2,61 +2,124 @@
 
 use lazy_static::lazy_static;
 
-use crate::{prelude::*, sync::SpinLock, task::Task};
+use crate::{cpu::this_cpu, cpu_local, prelude::*, sync::SpinLock, task::Task};
 
 lazy_static! {
-    pub(crate) static ref GLOBAL_SCHEDULER: SpinLock<GlobalScheduler> =
-        SpinLock::new(GlobalScheduler { scheduler: None });
+    /// The global scheduler is responsible for managing tasks across all CPUs in the system.
+    /// It provides a centralized point of task management, and all tasks will be put into the
+    /// queue of the global scheduler first when they are ready to run. The global scheduler can decide
+    /// the distribution and migration of tasks according to the CPU load to achieve the best resource utilization.
+    static ref GLOBAL_SCHEDULER: SpinLock<GeneralScheduler> =
+        SpinLock::new(GeneralScheduler::new());
+}
+
+cpu_local! {
+    /// Each CPU in the system has its own local scheduler instance, allowing for tasks to be
+    /// managed on a per-CPU basis. This is useful for tasks that are affine to a specific CPU
+    /// or for load balancing purposes where tasks are distributed among CPUs to avoid contention.
+    /// The local scheduler can optimize task execution by reducing task migration between CPUs and
+    /// minimizing synchronization overhead for task management. In fact, the most important thing
+    /// is that our current context switch mechanism will have a difficult competition problem in
+    /// the global scheduler.
```

Comment:
Just state the race we found here, try not to conceal your knowledge:
```suggestion
    /// minimizing synchronization overhead for task management. Furthermore, If a preempted task
    /// is emplaced back immediately to the global scheduler before the context switch, other processors
    /// may fetch the task with a stale context. And we couldn't make the progress between emplacing
    /// and context storing atomic.
```